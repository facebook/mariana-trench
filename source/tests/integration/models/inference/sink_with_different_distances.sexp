(method (private) "LFlow;.into_sink:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0 v1) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
(method (private) "LFlow;.into_into_sink:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0 v1) "LFlow;.into_sink:(LData;)V")
  (return-void)
 )
)
(method (private) "LFlow;.different_lengths:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0 v1) "LFlow;.into_sink:(LData;)V")
  (invoke-virtual (v0 v1) "LFlow;.into_into_sink:(LData;)V")
  (return-void)
 )
)
(method (private) "LFlow;.into_different_lengths:(LData;)V"
 (
  (load-param v0)
  (load-param v1)

  LINE(1)
  (invoke-virtual (v0 v1) "LFlow;.different_lengths:(LData;)V")
  (return-void)
 )
)
