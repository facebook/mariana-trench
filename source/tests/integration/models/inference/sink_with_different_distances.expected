{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "method" : "LFlow;.different_lengths:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LFlow;.into_into_sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 3,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        },
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LFlow;.into_sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 2,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "method" : "LFlow;.into_different_lengths:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LFlow;.different_lengths:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 3,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "method" : "LFlow;.into_into_sink:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LFlow;.into_sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 2,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "method" : "LFlow;.into_sink:(LData;)V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(1)",
          "distance" : 1,
          "kind" : "TestSink",
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    }
  ]
}
